package poseidon.adapter.expose_url;
import "poseidon_util.proto";

message UrlCoding
{
	optional bytes pid = 1;
	optional bytes dspid = 2;
	optional bytes bid	= 3;
	optional uint32 source= 4;
	optional bytes sid= 5;
	optional bytes deal_id= 6;
	optional bytes dev_id= 7;
	optional bytes aid= 8;
	optional bytes ad_id= 9;
	optional bytes creative_id= 10;
	optional uint32 advertiser_id= 11;
	optional uint32 campaign_id= 12;
	optional int32 bid_price= 13;
	optional int32 pdb_price= 14;
	optional int32 cost_price= 15;
	optional bytes billing_type= 16;
	optional int32 action= 17;
	optional int64 campaign_daily_budget= 18;
	optional uint32 smooth_type= 19;
	optional uint32 post_hours= 20;
	optional bytes acookie= 21;
	optional int32 org_price= 22;
	optional uint64 timestamp= 23;
	optional uint32 advertiser_daily_budget= 24;
	optional uint32 traffic_bid_flag= 25;
	optional int64 freq = 26;
	optional bytes creative_material_id= 27;
	optional bytes ch = 28;
	optional bytes trace_id = 29;
	optional int32 view_type= 30;
	optional int32 premium_rate= 31;
	optional bytes exp_id= 32;
	optional uint32 gid= 33;
	repeated poseidon.util.KeyValue algo_feedbacks = 34;
	optional uint32 os = 35;//1 android 2 ios
};

